File: app/src/main/java/com/backdoor/walcartandroidtest/View/Model/CategoryFragment/CategorySubCatRecyclerAdapter.java
package com.backdoor.walcartandroidtest.View.Model.CategoryFragment;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.backdoor.walcartandroidtest.R;
import com.example.GetCategoriesListQuery;

import java.util.List;


public class CategorySubCatRecyclerAdapter extends RecyclerView.Adapter<CategorySubCatRecyclerAdapter.MyViewHolder> {

    private List<GetCategoriesListQuery.Category> contact;
    private Context mContext;

    public CategorySubCatRecyclerAdapter(List<GetCategoriesListQuery.Category> contacts, Context context) {
        this.contact = contacts;
        this.mContext = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_category_sub_cat, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final MyViewHolder holder, final int position) {

        try {


        } catch (Exception ex) {
            Log.d("CategoryRootCat", "error: " + ex.getMessage());
        }

    }


    @Override
    public int getItemCount() {
        return 10;
    }


    static class MyViewHolder extends RecyclerView.ViewHolder {



        MyViewHolder(View itemView) {
            super(itemView);

        }
    }


    @SuppressLint("NotifyDataSetChanged")
    public void notifyChangeData(List<GetCategoriesListQuery.Category> dataList) {
        contact = dataList;
        notifyDataSetChanged();
    }

}

File: app/src/main/java/com/backdoor/walcartandroidtest/View/Model/CategoryFragment/CategorySubCatRecyclerAdapter.kt
package com.backdoor.walcartandroidtest.View.Model.CategoryFragment

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.backdoor.walcartandroidtest.R
import com.backdoor.walcartandroidtest.databinding.ItemCategorySubCatBinding
import com.example.GetCategoriesListQuery

class CategorySubCatRecyclerAdapter(
    private val mContext: Context,
    private var contact: List<GetCategoriesListQuery.Parent1?>
) :
    RecyclerView.Adapter<CategorySubCatRecyclerAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val binding = ItemCategorySubCatBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return MyViewHolder(binding)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        try {
            Log.d("CategoryRootCat", "SubCat: $contact")

            holder.binding.itemCategorySubCatTitle.text = contact[position]?.enName ?: ""

            holder.binding.itemCategorySubCatLayout.setOnClickListener {
                if (holder.binding.itemCategorySubCatDetails.visibility == View.GONE){
                    holder.binding.itemCategorySubCatDetails.visibility = View.VISIBLE
                    holder.binding.itemCategorySubCatTitleIcon.setImageResource(R.drawable.ic_baseline_keyboard_arrow_up_24)
                    holder.binding.itemCategorySubCatTitleIcon.setColorFilter(ContextCompat.getColor(mContext,R.color.catSelectedColor))
                    holder.binding.itemCategorySubCatTitle.setTextColor(mContext.resources.getColor(R.color.catSelectedColor))
                }else{
                    holder.binding.itemCategorySubCatDetails.visibility = View.GONE
                    holder.binding.itemCategorySubCatTitleIcon.setImageResource(R.drawable.ic_baseline_keyboard_arrow_down_24)
                    holder.binding.itemCategorySubCatTitleIcon.setColorFilter(ContextCompat.getColor(mContext,R.color.black_status_bar))
                    holder.binding.itemCategorySubCatTitle.setTextColor(mContext.resources.getColor(R.color.black_status_bar))
                }
            }

        } catch (ex: Exception) {
            Log.d("CategoryRootCat", "error: " + ex.message)
        }
    }

    override fun getItemCount(): Int {
        return if (contact.isEmpty()) {
            0
        } else {
            contact.size
        }
    }

    class MyViewHolder(val binding: ItemCategorySubCatBinding) : RecyclerView.ViewHolder(binding.root)

}
